src/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/Inlined/NullableDecimalAverageAggregationOperator.cs

```diff
@@ -118,8 +118,11 @@ protected override bool MoveNextCore(ref Pair<decimal, long> currentElement)
 
                     if (current.HasValue)
                     {
-                        sum += current.GetValueOrDefault();
-                        count++;
+                        checked
+                        {
+                            sum += current.GetValueOrDefault();
+                            count++;
+                        }
```

Comment:
> Do you mean only checking the count for float/double/decimal?

Yup.

> (I'm assuming you don't mean only checking the addition for long, which is the only thing that can overflow at the moment?

I have the same nagging thoughts, though it also feels wrong to penalize the generated code because of something that could theoretically change in the future.  Were you able to verify that the addition of checked isn't measurable on the resulting performance, due to the other overheads involved?